cinema/permissions.py

```diff
@@ -0,0 +1,26 @@
+from django.http import Http404
+from rest_framework.permissions import BasePermission
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+
+    def has_permission(self, request, view):
+        return bool(
+            (
+                view.action in ["retrieve", "list"]
+                and request.user
+                and request.user.is_authenticated
+            )
+            or (request.user and request.user.is_staff)
+        )
+
+
+class OrderPermission(BasePermission):
```

Comment:
There is no need for this permission class, having `IsAdminOrIfAuthenticatedReadOnly` is sufficient